digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala

```diff
@@ -301,6 +328,92 @@ class AssetDataImporter {
+    val eventBus = new DummyEventBus
+    val linkService = new RoadLinkService(vvhClient, eventBus, new DummySerializer)
+
+    val roadsWithMultipleValues = conversionDatabase.withDynSession {
+      sql"""SELECT t.tie, t.aosa, ajr, greatest(aet,let) as addrM, tietyyppi as type_before, (SELECT tietyyppi FROM VVH_TIEOSOITE_NYKY u WHERE u.tie = t.tie AND u.aosa=t.aosa AND (u.ajr = t.ajr OR u.ajr=0 OR t.ajr=0)
+            AND ((u.aet = t.let and t.aet < t.let) or (u.let = t.aet and t.aet > t.let) ) AND ROWNUM < 2) type_after, ely FROM VVH_TIEOSOITE_NYKY t
+            JOIN (SELECT gr.TIE, gr.AOSA, count(distinct TIETYYPPI) FROM VVH_TIEOSOITE_NYKY gr group by gr.tie, gr.aosa having count(distinct TIETYYPPI) > 1) foo ON
+            (foo.tie = t.tie AND foo.aosa = t.aosa)
+            WHERE (  EXISTS (SELECT 1 FROM VVH_TIEOSOITE_NYKY chg WHERE chg.tie = t.tie and chg.aosa=t.aosa and
+            (chg.ajr = t.ajr OR chg.ajr=0 OR t.ajr=0) AND  ((chg.aet = t.let and t.aet < t.let) or (chg.let = t.aet and t.aet > t.let) )
+            AND chg.tietyyppi != t.tietyyppi)) order by tie, aosa, addrM, ajr
+         """.as[(Long, Long, Long, Long, Int, Int, Int)].list.map {
+        case (roadNumber, roadPartNumber, trackCode, addressMChangeRoadType, roadTypeBefore, roadTypeAfter, ely) =>
+          RoadTypeChangePoints(roadNumber, roadPartNumber, addressMChangeRoadType, RoadType.apply(roadTypeBefore), RoadType.apply(roadTypeAfter), ely)
+      }
+    }
+
+    val roadsWithSingleRoadType = conversionDatabase.withDynSession {
+      sql""" select distinct tie, aosa, tietyyppi, ely from VVH_TIEOSOITE_NYKY
+             where (tie, aosa) in(
+             SELECT gr.TIE, gr.AOSA FROM VVH_TIEOSOITE_NYKY gr  group by gr.tie, gr.aosa  having count(distinct gr.TIETYYPPI) = 1)
+              union
+            select distinct tie, aosa, tietyyppi, ely from vvh_tieosoite_taydentava
+            where (tie, aosa) in(
+            SELECT t.TIE, t.AOSA FROM vvh_tieosoite_taydentava t  group by t.tie, t.aosa  having count(distinct t.TIETYYPPI) = 1)
+            order by tie, aosa
+        """.as[(Long, Long, Long, Long)].list
+    }
+
+    withDynTransaction{
+      roadsWithSingleRoadType.foreach(road => {
+        updateRoadWithSingleRoadType(road._1, road._2, road._3, road._4)
+      })
+
+      roadsWithMultipleValues.foreach(road =>{
+        val addresses = RoadAddressDAO.fetchByRoadPart(road.roadNumber, road.roadPartNumber, true)
+        addresses.nonEmpty match {
+          case true => {
+            println(s"updating tie = ${road.roadNumber}, aosa = ${road.roadPartNumber}")
+            addresses.foreach(address =>{
+              roadType(roadsWithMultipleValues.filter { r=> r.roadNumber == road.roadNumber && r.roadPartNumber == road.roadPartNumber }, address) match {
+                case Left(roadType) =>{
+                  sqlu"""UPDATE ROAD_ADDRESS SET ROAD_TYPE = ${roadType.value}, ELY= ${road.elyCode} where ID = ${address.id} """.execute
```

Comment:
Could be better to do updates in bigger chunks than one row at a time.

Reply:
Not worth it, too low number of rows affected -> performance increase is negligible.